fork-off/src/account_setting.rs

```diff
@@ -0,0 +1,66 @@
+use std::{collections::HashMap, str::FromStr};
+
+use crate::Storage;
+use codec::Encode;
+use log::info;
+use serde::Deserialize;
+
+use crate::types::{AccountId, Balance, StorageKey, StoragePath, StorageValue};
+
+/// This struct is copied and type-specialized from `pallet_balances::AccountData`.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Deserialize, Encode, Default)]
+pub struct AccountData {
+    pub free: Balance,
+    pub reserved: Balance,
+    pub misc_frozen: Balance,
+    pub fee_frozen: Balance,
+}
+
+/// This struct is copied and type-specialized from `frame_system::AccountInfo`.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Deserialize, Encode, Default)]
+pub struct AccountInfo {
+    pub nonce: u32,
+    pub consumers: u32,
+    pub providers: u32,
+    pub sufficients: u32,
+    pub data: AccountData,
+}
+
+impl From<AccountInfo> for StorageValue {
+    fn from(account_info: AccountInfo) -> StorageValue {
+        StorageValue::new(&hex::encode(Encode::encode(&account_info)))
+    }
+}
+
+impl AccountInfo {
+    /// Create `AccountInfo` with all parameters set to `0` apart from free balances, which is
+    /// set to `free` and number of providers, which is set to `1`.
+    pub fn from_free(free: Balance) -> Self {
+        Self {
+            providers: 1,
+            data: AccountData {
+                free,
+                ..AccountData::default()
+            },
+            ..AccountInfo::default()
+        }
+    }
+}
+
+pub type AccountSetting = HashMap<AccountId, AccountInfo>;
+
+fn get_account_map() -> StoragePath {
+    StoragePath::from_str("System.Account").unwrap()
+}
+
+pub fn apply_account_setting(mut state: Storage, setting: AccountSetting) -> Storage {
+    let account_map: StorageKey = get_account_map().into();
+    for (account, info) in setting {
+        let account_hash = account.clone().into();
+        let key = &account_map.join(&account_hash);
+
+        state.insert(key.clone(), info.clone().into());
+        info!("Account info of `{:?}` set to `{:?}`", account, info);
```

Comment:
Please add `target`

Reply:
We can add  `target` here since we're adding/modifying this line anyway,  but we don't need to modify all `log!` lines in this PR.

Reply:
okay, done